src/docker/composeappengine.cc

```diff
@@ -143,7 +143,15 @@ bool ComposeAppEngine::isRunning(const App& app) const {
   bool started_state = false;
   try {
     AppState state(app, appRoot(app));
-    started_state = state() == AppState::State::kStarted;
+    if (app.uri == state.version()) {
+      started_state = state() == AppState::State::kStarted;
+    } else {
+      // The state file exists but it describes a state of some other App not the one
+      // that specified via the param. It can happen if a new Target App download or start fails
+      // and aklite checks if the current App is running
+      LOG_DEBUG << "Failed to get/set App state, fallback to checking the dockerd state";
+      started_state = true;
```

Comment:
i'm not quite understanding why we don't do `started_state = state() == AppState::State::kStarted;` here also?

Reply:
Because a client calling isRunning() is asking about the App a version of which (current target) differs from a version of App (new Target) state of which is a recorder/stored in the state file.
It can happen when aklite tried to download a new Target's App and failed or a download succeeded but start failed and after that somehow isRunning is called for a current Target, not a new one. Or if a rollback happens after a reboot then the state file refers to a state of the new Target App while the liteclient asks if running the current App... 
also, this verification https://github.com/foundriesio/aktualizr-lite/blob/94577e0300948069912993984dab021849a3c9cb/tests/aklite_test.cc#L93 will fail if we do it.